rpcserver.go

```diff
@@ -1867,6 +1892,296 @@ func (r *rpcServer) AnchorVirtualPsbts(ctx context.Context,
 	}, nil
 }
 
+// CommitVirtualPsbts creates the output commitments and proofs for the given
+// virtual transactions by committing them to the BTC level anchor transaction.
+// In addition, the BTC level anchor transaction is funded and prepared up to
+// the point where it is ready to be signed.
+func (r *rpcServer) CommitVirtualPsbts(ctx context.Context,
+	req *wrpc.CommitVirtualPsbtsRequest) (*wrpc.CommitVirtualPsbtsResponse,
+	error) {
+
+	if len(req.VirtualPsbts) == 0 {
+		return nil, fmt.Errorf("no virtual PSBTs specified")
+	}
+
+	pkt, err := psbt.NewFromRawBytes(bytes.NewReader(req.AnchorPsbt), false)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding packet: %w", err)
+	}
+
+	activePackets := make([]*tappsbt.VPacket, len(req.VirtualPsbts))
+	for idx := range req.VirtualPsbts {
+		activePackets[idx], err = tappsbt.Decode(req.VirtualPsbts[idx])
+		if err != nil {
+			return nil, fmt.Errorf("error decoding virtual "+
+				"packet at index %d: %w", idx, err)
+		}
+	}
+
+	passivePackets := make([]*tappsbt.VPacket, len(req.PassiveAssetPsbts))
```

Comment:
does this mean that the user has to be aware & responsible of which assets are being passively carried when creating a vPSBT?

Reply:
For the new flow (`FundVirtualPsbts -> CommitVirtualPsbts -> PublishAndLogTransfer`) yes. For any existing flow, no.

Reply:
Do we have any measures in place to protect a user from accidentally urning assets if they don't pass them in properly to `CommitVirtualPsbts`?